### PR TITLE
Networking: speed up log download

### PR DESCRIPTION
**Pull request by tridge**
This speeds up log download over networking by allowing for larger than 4 reads at a time if the memory region is suitable
Speedup on CubeRed is log download goes from around 1.3MByte/s to around 2.4MByte/s
On PPP the speed goes from around 600kbyte/s to about 750kbyte/s

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c [hunk 1]
```diff
@@ -616,3 +616,41 @@ void ff_memfree(void* mblock)
     free(mblock);
 }
 #endif // USE_POSIX
+
+/*
+  return true if a memory region is safe for a DMA operation
+ */
+bool mem_is_dma_safe(const void *addr, uint32_t size, bool filesystem_op)
```
Comment (magicrub) [inline libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c:623]: This function could be const

File: libraries/AP_Networking/AP_Networking_Config.h [hunk 2]
```diff
@@ -117,5 +117,5 @@
 #endif
 
 #ifndef AP_NETWORKING_SENDFILE_BUFSIZE
-#define AP_NETWORKING_SENDFILE_BUFSIZE 10000
+#define AP_NETWORKING_SENDFILE_BUFSIZE (64*512)
```
Comment (magicrub) [inline libraries/AP_Networking/AP_Networking_Config.h:120]: Why this grouping instead of 32*1024?

**Review by magicrub — COMMENTED** [review]
I don't see anything obviously wrong but I didn't dive in to the details to confirm all those addresses are correct

File: libraries/AP_Networking/AP_Networking_Config.h (hunk 2 above)
Comment (tridge) [reply libraries/AP_Networking/AP_Networking_Config.h:120]: sector size is 512 bytes, we need multiple of sector size

File: libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c:623]: no it can't be, this is C not C++

**Review by magicrub — APPROVED** [review]

**Pull request merged**